app/wizards/schools/decorated_school.rb

```diff
@@ -0,0 +1,11 @@
+module Schools
+  class DecoratedSchool < SimpleDelegator
+    def latest_registration_choices(contract_period: ContractPeriod.containing_date(Time.zone.today))
+      @latest_registration_choices ||= Schools::LatestRegistrationChoices.new(school: self, contract_period:)
```

Comment:
Will the memoisation cause `@latest_registration_choices` to  return stale results if a different `contract_period` is passed in? 

I mean it's not much of an issue since contract period doesn't change much but wanted to flag.